test/integration/targets/json_cleanup/module_output_cleaning.yml

```diff
@@ -0,0 +1,25 @@
+- name: ensure we clean module output well
+  hosts: localhost
+  gather_facts: false
+  tasks:
+    - name: call module that spews extra stuff
+      bad_json:
+      register: clean_json
+      ignore_errors: true
+
+    - name: all expected is there
+      assert:
+        that:
+          - clean_json is success
+          - clean_json is not changed
+          - "clean_json['msg'] == 'good json response'"
+
+    - name: all non wanted is not there
+      assert:
+        that:
+          - item not in clean_json.values()
+      loop:
+         - this stuff should be ignored
+         - [ looks like a json list]
```

Comment:
Maybe also
```suggestion
         - ' looks like a json list'
```

Reply:
made additional and not substitute, as parser was passing a list up

Reply:
Ah, I meant this to be an addition, looks like I screwed that suggestion up :)